### ISSUE TITLE
Firebase auth: Operation timeout issues on login, using social media login eg. google

### ISSUE THREAD

**Issue by vaibhav-pande** [firebase/firebase-js-sdk#4040]
### [REQUIRED] Describe your environment
  * Operating System version: Windows 10
  * Browser version: Chrome 
  * Firebase SDK version: 7.19.1
  * Firebase Product: auth
 
 ### [REQUIRED] Describe the problem
We are using firebase on webapp which acts as an SSO provider for our connected applications, deployed on appengine.
At times, after we enter the credentials for social media login eg. google, it doesn't move ahead and gets stuck, **giving operation time out error.**
When we inspected on console, we could see the error with iframewrapper.js which is part of the highlighted javascript.(iframewrapper.js)
**Even though having good network bandwidth, many users are facing this issue they are able to login only after multiple retries.**
**What is the probable cause for this issue and solution?
Is this issue resolved post 7.19.1 Firebase release? We have continuously updated the versions from 17.13.2 till 7.19.1.**
We could see many issues logged on various forums regarding operation time out, but we couldn't get satisfactory resolution till date.

#### Steps to reproduce:
1) Go to sso sign in page
2) Click on social media login option eg. google
3) Enter credentials
4) It comes back to sso sign in page giving operation time out error on google icon

Attaching screenshot for reference:
![image](https://user-images.githubusercontent.com/46163092/98247827-215b7a80-1f9a-11eb-9f14-51467b9e436d.png)

![image](https://user-images.githubusercontent.com/46163092/98247897-3a642b80-1f9a-11eb-8a6d-6f29d83c4749.png)

**Comment by samhorlbeck** [comment]
Hi, thanks for reaching out--

A couple quick questions to help debugging:
 1. Are you using `signInWithPopup()` or `signInWithRedirect()`? (From your post it sounds like redirect but I want to double check)
 2. Does the network tab show any connections failing?

**Comment by vaibhav-pande** [comment]
> Hi, thanks for reaching out--
> 
> A couple quick questions to help debugging:
> 
> 1. Are you using `signInWithPopup()` or `signInWithRedirect()`? (From your post it sounds like redirect but I want to double check)
> 2. Does the network tab show any connections failing?

Thanks for response, please see point wise answers below:
1) Yes, using signInWithRedirect()
2) Connections are **not** failing in network tab, if we still anything on this we will update.

**Comment by samhorlbeck** [comment]
Thanks for that info. This does sound like a bug; I've filed one internally to track (b/172836836).

Just a few more questions:
Is this happening for all Chrome? Just incognito? Are you able to reproduce this in another browser (Firefox/etc) as well?

**Comment by samhorlbeck** [comment]
Does the iframe show up in the DOM (meaning it was successfully loaded)?

**Comment by vaibhav-pande** [comment]
> Thanks for that info. This does sound like a bug; I've filed one internally to track (b/172836836).
> 
> Just a few more questions:
> Is this happening for all Chrome? Just incognito? Are you able to reproduce this in another browser (Firefox/etc) as well?

Thanks for response and filing bug. 
Yes we are facing this issue on other browsers as well not just Chrome/Incognito

**Comment by vaibhav-pande** [comment]
> Does the iframe show up in the DOM (meaning it was successfully loaded)?

Yes as per our understanding, it is getting loaded, please refer screenshot 2.

**Comment by vaibhav-pande** [comment]
Few more screenshot for reference:
![image](https://user-images.githubusercontent.com/46163092/98804421-2dcd5080-243c-11eb-8c4b-57c61b5ebf82.png)

![image](https://user-images.githubusercontent.com/46163092/98804444-36be2200-243c-11eb-8806-65ed17088103.png)

**Comment by vaibhav-pande** [comment]
Could you please let us know any update on this? @samhorlbeck

**Comment by mattread90** [comment]
Our users are also experiencing this problem, or at least a variation of it.

We are using `signInWithPopup()`, but are getting the same `Network Error` triggered by the timeout. It seems to be happening across all browsers, to hundreds of different users each day.

For the users who experience this, it seems to be consistent, however we have never been able to recreate the problem ourselves. Having seen a screen recording of one user, it seems the popup opens but remains blank (with `about:blank` in the URL bar), before closing after about 5 seconds.

**Comment by blikblum** [comment]
Same here. Using `signInWithPopup` . Getting Network Error. The popup opens, stays blank and closes (see video).

It occurs in a specific computer with both Firefox and Chrome

https://user-images.githubusercontent.com/3047126/118903482-4425fe80-b8ee-11eb-8cd0-2f129e6307ba.mp4

**Comment by FabianSellmann** [comment]
Also seeing similar reports from users and in our error logging. We are using `signInWithRedirect()`. I could not reproduce it myself yet, but from the error logs it seems like there is not even a fetch request fired to any google/firebase api before the `Network Error` is thrown.

**Comment by sam-gc** [comment]
Hi everyone, thanks for the added info. It sounds like the intermediary page isn't loading (between your app and social login; you'll notice, when it works, you're briefly redirected to `<your-auth-domain>/__/auth/<lots of stuff>`). In all these cases, it appears the `/__/auth` handlers aren't loading.

Are you using Firebase Hosting? @blikblum, since you have a computer that you can reproduce this on, would you mind checking to see if you can connect manually to "your-domain.web.app/__/auth/handler"? (Ignore the content of this page if does load, I just want to know if the page is available/loading at all)

**Comment by blikblum** [comment]
I can try. It's a customer computer
